### PR TITLE
Upgrade to JDK 11

### PR DESCRIPTION
**Pull request by shaikhu**

### CODE REVIEW

File: gradle/integration-test.gradle [hunk 1]
```diff
@@ -12,10 +12,7 @@ task integrationTest(type: Test) {
   group = 'verification'
 }
 
-['**/ScanIT_Gradle_Versions_3_3*.class',
- '**/ScanIT_Gradle_Versions_4_1*.class',
- '**/ScanIT_Gradle_Versions_4_4*.class',
- '**/ScanIT_Gradle_Versions_5_0*.class',
```
Comment (shaikhu) [inline gradle/integration-test.gradle:18]: These versions of gradle aren't compatible with Java 11.

**Comment by shaikhu** [comment]
Will add new gradle versions for integration testing in a separate branch/pr

File: build.gradle [hunk 2]
```diff
@@ -64,8 +64,7 @@ dependencies {
   testImplementation gradleTestKit()
   testImplementation "junit:junit:$junitVersion"
   testImplementation "org.assertj:assertj-core:$assertJVersion"
-  testImplementation "org.powermock:powermock-module-junit4:$powermockVersion"
-  testImplementation "org.powermock:powermock-api-mockito2:$powermockVersion"
+  testImplementation "org.mockito:mockito-core:${mockitoVersion}"
```
Comment (shaikhu) [inline build.gradle:67]: Switched from powermock to mockito due to powermock/jdk 11 issues e.g. the test `NexusIqScanTaskTest.testScan_realWithScanTargets` would print warnings for illegal reflective access. The approach taken with mockito does not.

File: gradle.properties [hunk 3]
```diff
@@ -24,5 +24,5 @@ logbackVersion=1.3.14
 commonsIoVersion=2.16.1
 
 junitVersion=4.13.2
-powermockVersion=2.0.9
+mockitoVersion=5.12.0
```
Comment (shaikhu) [inline gradle.properties:27]: Switched from powermock to mockito due to powermock/jdk 11 issues e.g. the test `NexusIqScanTaskTest.testScan_realWithScanTargets` would print warnings for illegal reflective access. The approach taken with mockito does not.

File: gradle/integration-test.gradle (hunk 1 above)
Comment (guillermo-varela) [reply gradle/integration-test.gradle:18]: > These versions of gradle aren't compatible with Java 11. Plan is to add new/supported gradle versions for integration testing in a separate branch/pr (as I expect issues while adding them)

This means we would also need to update the `Compatibility` section in the README file

File: gradle.properties (hunk 3 above)
Comment (guillermo-varela) [reply gradle.properties:27]: We were already using Mockito through `org.powermock:powermock-api-mockito2` so now using just Mockito, given that it supports mocking static methods, seems ok.

File: gradle/integration-test.gradle (hunk 1 above)
Comment (guillermo-varela) [reply gradle/integration-test.gradle:18]: > Plan is to add new/supported gradle versions for integration testing in a separate branch/pr (as I expect issues while adding them)

While at it, we might also want/need to update the Gradle version we're using to build the plugin.

Comment (guillermo-varela) [reply gradle/integration-test.gradle:18]: Just bumped up the project version as this is a breaking change:
d023b801052630f3d12bb8abc8fb32d068775d40

Comment (shaikhu) [reply gradle/integration-test.gradle:18]: > > Plan is to add new/supported gradle versions for integration testing in a separate branch/pr (as I expect issues while adding them)
> 
> While at it, we might also want/need to update the Gradle version we're using to build the plugin.

Agreed we probably need to update it, but I'd prefer to do that in another PR. The current version works with JDK 11.

Comment (shaikhu) [reply gradle/integration-test.gradle:18]: > This means we would also need to update the `Compatibility` section in the [README file](https://github.com/sonatype-nexus-community/scan-gradle-plugin?tab=readme-ov-file#compatibility):

👍 d970155d
 
> It probably be useful to mention the latest version where older Gradle versions where supported and the version where Gradle 5 is the lowest supported version with Java 11.

👍 795e49f7

**Pull request merged**